### PR TITLE
Adversarial Principal Component Analysis

### PR DESCRIPTION
**Pull request by austinTalbot7241993**
We can use this model to make the principal components orthogonal to sensitive variables such as race or sex.

### CODE REVIEW

File: python/python/bystro/supervised_ppca/gf_adversarial_pt.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+        Parameters
+        ----------
+        X : NDArray,(n_samples,p)
+            The data
+
+        Returns
+        -------
+        W_ : torch.tensor,shape=(n_components,p)
+            The loadings of our latent factor model
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_adversarial_pt.py:290]: Do we want to keep documentation specific to factor models?

File: python/python/bystro/supervised_ppca/gf_adversarial_pt.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    n_components : int
+        The latent dimensionality
+
+    prior_options : dict
+        The hyperparameters for the prior on the parameters
+
+    mu : float>0,default=1.0
+        The adversarial penalty
+
+    gamma : float,default=10.0
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_adversarial_pt.py:91]: missing the training options parameter description

**Review by akotlar — COMMENTED** [review]
very minor comments

File: python/python/bystro/supervised_ppca/gf_adversarial_pt.py (hunk 1 above)
Comment (austinTalbot7241993) [reply python/python/bystro/supervised_ppca/gf_adversarial_pt.py:290]: What do you mean?

**Pull request merged**